### PR TITLE
Filter refactor

### PR DESCRIPTION
**Pull request by ibmibmibm**

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build? (admin please type: ok to test)

**Comment by Kokan** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build? (admin please type: ok to test)

File: lib/filter/filter-op.c [hunk 1]
```diff
@@ -34,10 +34,12 @@ fop_init(FilterExprNode *s, GlobalConfig *cfg)
...
-    self->left->init(self->left, cfg);
-  if (self->right && self->right->init)
-    self->right->init(self->right, cfg);
+  g_assert(self->left);
+  g_assert(self->right);
+
+  filter_expr_init(self->left, cfg);
+  filter_expr_init(self->right, cfg);
+
   self->super.modify = self->left->modify || self->right->modify;
```
Comment (bazsi) [inline lib/filter/filter-op.c]: much nicer, thanks.

File: lib/filter/filter-pipe.c [hunk 2]
```diff
@@ -35,7 +35,7 @@ log_filter_pipe_init(LogPipe *s)
   LogFilterPipe *self = (LogFilterPipe *) s;
   GlobalConfig *cfg = log_pipe_get_config(s);
 
-  filter_expr_init(self->expr, log_pipe_get_config(s));
+  filter_expr_init(self->expr, cfg);
   if (!self->name)
```
Comment (bazsi) [inline lib/filter/filter-pipe.c]: good one.

File: lib/filter/filter-pri.c [hunk 3]
```diff
@@ -48,7 +48,7 @@ filter_facility_eval(FilterExprNode *s, LogMessage **msgs, gint num_msg)
     {
       return !!(self->valid & (1 << fac_num)) ^ self->super.comp;
     }
-  return self->super.comp;
+  g_assert_not_reached();
 }
```
Comment (bazsi) [inline lib/filter/filter-pri.c]: makes sense again. so all of these look good to me.

**Comment by Kokan** [comment]
@kira-syslogng retest this please

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Pull request merged**